joblib/numpy_pickle.py

```diff
@@ -92,6 +95,17 @@ def write_array(self, array, pickler):
             # pickle protocol.
             pickle.dump(array, pickler.file_handle, protocol=2)
         else:
+            try:
+                current_pos = pickler.file_handle.tell()
+                alignment = current_pos % NUMPY_ARRAY_ALIGNMENT_BYTES
+
+                if alignment != 0:
+                    padding = b' ' * (NUMPY_ARRAY_ALIGNMENT_BYTES - alignment)
+                    pickler.file_handle.write(padding)
+            except io.UnsupportedOperation:
+                # TODO log something somewhere?
```

Comment:
If people plan to memmap the resulting pickle, they probably dump to a real filesystem-backed handle. So it might be fine to process silently otherwise.